mindsdb/integrations/handlers/sentence_transformer_handler/settings.py

```diff
@@ -0,0 +1,79 @@
+from functools import lru_cache
+from typing import List, Union
+
+import pandas as pd
+import torch
+from pydantic import BaseModel, Field
+from sentence_transformers import SentenceTransformer
+
+
+@lru_cache()
+def load_embeddings_model(embeddings_model_name):
+    try:
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        embedding_model = SentenceTransformer(
+            model_name_or_path=embeddings_model_name, device=device
+        )
+    except ValueError:
+        raise ValueError(
+            f"The {embeddings_model_name}  is not supported, please select a valid option from Hugging Face Hub!"
+        )
+    return embedding_model
+
+
+class Document(BaseModel):
+    """
+    store for text, capturing metadata
+    """
+
+    text: str
+    metadata: Field(default_factory=dict)
+
+
+class DfLoader:
+    def __init__(self, data_frame: pd.DataFrame, page_content_column: str):
+        self._data_frame = data_frame
+        self._page_content_column = page_content_column
+
+    def load(self) -> List[Document]:
+        """takes a Dataframe column and loads it as a list of documents"""
+        documents = []
+        for n_row, frame in self._data_frame[self._page_content_column].iteritems():
```

Comment:
Potentially slow for large frames, but fine for v1.0. Just highlighting for future reference.

Reply:
tbh only reason for this was I thought we wanted to log metadata, if we don't then can remove and make faster

Reply:
Ah I see, let's keep like this for now then